code/modules/power/gravitygenerator.dm

```diff
@@ -440,6 +440,15 @@ GLOBAL_LIST_EMPTY(gravity_generators) // We will keep track of this by adding ne
 		setting = value
 		shake_everyone()
 
+/obj/machinery/gravity_generator/main/proc/blackout()
+	charging_state = POWER_IDLE
+	on = FALSE
+	charge_count = 0
+	breaker = FALSE
+	set_power()
+	disable()
+	investigate_log("has turned down by blackout event.", INVESTIGATE_GRAVITY)
```

Comment:
```suggestion
	investigate_log("was turned off by blackout event.", INVESTIGATE_GRAVITY)
```
What if this is turned off by anomaly instead of the event?
Also, disable() takes care of on and charging_state.

Reply:
I removed `disable()` and `charging_state`, also made it clearer in the log that the gravity failure was because of the blackout event/anomaly detonation and not player action.